libpod/container_internal.go

```diff
@@ -131,14 +133,92 @@ func (c *Container) AttachSocketPath() string {
 	return filepath.Join(c.runtime.ociRuntime.socketsDir, c.ID(), "attach")
 }
 
+// exitFilePath gets the path to the container's exit file
+func (c *Container) exitFilePath() string {
+	return filepath.Join(c.runtime.ociRuntime.exitsDir, c.ID())
+}
+
+// create a bundle path and associated files for an exec session
+func (c *Container) createExecBundle(sessionID string) (err error) {
+	bundlePath := c.execBundlePath(sessionID)
+	if createErr := os.MkdirAll(bundlePath, execDirPermission); createErr != nil {
+		return createErr
+	}
+	defer func() {
+		if err != nil {
+			if err2 := os.Remove(bundlePath); err != nil {
+				logrus.Warnf("error removing exec bundle after creation caused another error: %v", err2)
+			}
+		}
+	}()
+	f, err2 := os.OpenFile(c.execLogPath(sessionID), os.O_RDONLY|os.O_CREATE, execFilePermission)
+	if err2 != nil {
+		err = errors.Wrapf(err2, "Error creating file for container %s exec session %s", c.ID(), sessionID)
+		return
+	}
+	if err2 := f.Close(); err2 != nil {
+		logrus.Warnf("Error in closing newly created log path file: %v", err2)
+	}
+	if err2 := os.MkdirAll(c.execExitFileDir(sessionID), execDirPermission); err2 != nil {
+		// The directory is allowed to exist
+		if !os.IsExist(err2) {
+			err = errors.Wrapf(err2, "error creating OCI runtime exit file path %s", c.execExitFileDir(sessionID))
+		}
+	}
+	return
+}
+
+// cleanup an exec session after its done
+func (c *Container) cleanupExecBundle(sessionID string) error {
+	return os.RemoveAll(c.execBundlePath(sessionID))
+}
+
+// the path to a containers exec session bundle
+func (c *Container) execBundlePath(sessionID string) string {
```

Comment:
We ought to throw all these one-liner path helpers in a separate file... Tracking them down is getting annoying

Reply:
Do you want that here or in a separate PR?

Reply:
Yeah let's not make this one bigger than it already is, separate PR is fine